app/workers/functions.php

```diff
@@ -80,25 +84,45 @@
 
         $runtime = $runtimes[$function->getAttribute('runtime')];
 
+        $headers['x-appwrite-trigger'] = $trigger;
+        $headers['x-appwrite-event'] = $event ?? '';
+        $headers['x-appwrite-user-id'] = $user->getId() ?? '';
+        $headers['x-appwrite-user-jwt'] = $jwt ?? '';
+
         /** Create execution or update execution status */
         $execution = $dbForProject->getDocument('executions', $executionId ?? '');
         if ($execution->isEmpty()) {
+            $headersFiltered = [];
+            foreach ($headers as $key => $value) {
+                if (\in_array(\strtolower($key), FUNCTION_ALLOWLIST_HEADERS_REQUEST)) {
+                    $headersFiltered[] = [ 'name' => $key, 'value' => $value ];
+                }
+            }
+
             $executionId = ID::unique();
-            $execution = $dbForProject->createDocument('executions', new Document([
+            $execution = new Document([
                 '$id' => $executionId,
                 '$permissions' => $user->isEmpty() ? [] : [Permission::read(Role::user($user->getId()))],
-                'functionId' => $functionId,
-                'functionInternalId' => $functionInternalId,
-                'deploymentInternalId' => $deploymentInternalId,
-                'deploymentId' => $deploymentId,
-                'trigger' => $trigger,
-                'status' => 'waiting',
-                'statusCode' => 0,
-                'response' => '',
-                'stderr' => '',
+                'functionInternalId' => $function->getInternalId(),
+                'functionId' => $function->getId(),
+                'deploymentInternalId' => $deployment->getInternalId(),
+                'deploymentId' => $deployment->getId(),
+                'trigger' => 'http',
+                'status' => 'processing',
+                'responseStatusCode' => 0,
+                'responseHeaders' => [],
+                'requestPath' => $path,
+                'requestMethod' => $method,
+                'requestHeaders' => $headersFiltered,
+                'errors' => '',
+                'logs' => '',
                 'duration' => 0.0,
-                'search' => implode(' ', [$function->getId(), $executionId]),
-            ]));
+                'search' => implode(' ', [$functionId, $executionId]),
+            ]);
+
+            if ($function->getAttribute('logging')) {
+                $execution = $dbForProject->createDocument('executions', $execution);
+            }
```

Comment:
What is the `logging` attribute? why are we creating an execution document based on the presence of this attribute?

Reply:
You are correct.
It can be toggled off for privacy/security, and performance.